src/ml/neural_net/image_augmentation.hpp

```diff
@@ -239,6 +239,13 @@ class image_augmenter {
     std::vector<std::vector<image_annotation>> annotations_batch;
   };
 
+  /** Data structure that contains result and labeled image. */
+  struct result_with_dimension {
```

Comment:
This type is not used or otherwise referenced anywhere in `image_augmenter.cpp`. Please move it to `object_detector` as a private struct

Reply:
Also worth considering having this inherit from `result` 

When moved to the OD code, though, could probably be called something like `inference_batch` or something like that. The name should give some indication of how it’s used. Here, the name `result_with_dimension` is not clear, since this type isn’t used as the result of any function defined in this file.

Reply:
this is nice.